R/entities_objects.R

```diff
@@ -0,0 +1,114 @@
+# Contains functions to parse the objects described here: 
+# https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/entities
+
+# <https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/entities#hashtags>
+hashtags <- function(x) {
+  if (NROW(x) == 0) {
+    return(data.frame(text = NA, indices = I(list(NA)), 
+                      stringsAsFactors = FALSE))
+  }
+  i <- indices_vec(x$indices)
+  data.frame(text = x$text, indices = I(i))
+}
+
+
+indices_vec <- function(x) {
+  i <- lapply(x, function(y){matrix(y, ncol = 2, 
+                                    dimnames = list(NULL, 
+                                                    c("start", "end")))})
+  i
+}
+
+# The extended entities is really for media
+# <https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/extended-entities>
+media <- function(x) {
+  df <- data.frame("id" = NA, "id_str" = NA, "indices" = I(list(NA)), 
+                   "media_url" = NA, "media_url_https" = NA, 
+                   "url" = NA, "display_url" = NA, "expanded_url" = NA, 
+                   "type" = NA, "sizes" = I(list(NA)), "ext_alt_text" = NA,
+                   stringsAsFactors = FALSE)
+  if (NROW(x) == 0) {
+    return(df)
+  }
+  indices <- as.data.frame(t(simplify2array(x$indices)))
+  colnames(indices) <- c("start", "end")
+  x$indices <- I(list(indices))
+  sizes <- rbind(x$sizes$large, x$sizes$small, x$sizes$thumb, x$sizes$medium)
+  sizes$type <- c("large", "small", "thumb", "medium")
+  x$sizes <- list(sizes)
+  x[setdiff(colnames(df), colnames(x))] <- rep(NA, nrow(x))
+  x
+}
+
+urls <- function(x) {
+  df <- data.frame("url" = NA, "expanded_url" = NA, "display_url" = NA, 
+                   "indices" = I(list(NA)), "unwound" = I(list(NA)), 
+                   stringsAsFactors = FALSE)
+  if (NROW(x) == 0) {
+    return(df)
+  }
+  indices <- as.data.frame(t(simplify2array(x$indices)))
+  colnames(indices) <- c("start", "end")
+  x$indices <- I(indices)
+  x[setdiff(colnames(df), colnames(x))] <- rep(NA, nrow(x))
+  x
+}
+
+# PowerTrack @ screen_name
+# has:mentions
+# <https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/entities#mentions>
+user_mentions <- function(x) {
+  df <- data.frame("screen_name" = NA, "name" = NA, "id" = NA, "id_str" = NA,
+                   "indices" = I(list(NA)), stringsAsFactors = FALSE)
+  if (NROW(x) == 0) {
+    return(df)
+  }
+  indices <- as.data.frame(t(simplify2array(x$indices)))
+  colnames(indices) <- c("start", "end")
+  x$indices <- indices
+  x[setdiff(colnames(df), colnames(x))] <- rep(NA, nrow(x))
+  rownames(x) <- NULL
+  x
+}
+
+# PowerTrack $ text
+# has:symbol
+# They are the same
+# <https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/entities#symbols>
+symbols <- hashtags
```

Comment:
Would be a bit simpler to understand if you just had one function.

Reply:
On `parse_entities2` I use the name of the columns to match with the appropriate function to parse it. So I needed a symbols function and it matched with the same of what hasthags function was doing so I created it that way.

Reply:
In that case, I'd put this function right next to `hashtags` and mention why you define it (i.e. what you said above) in a comment.